cli/internal/cmd/upgradecheck.go

```diff
@@ -227,11 +236,25 @@ func filterK8sUpgrades(currentVersion string, newVersions []string) []string {
 	return result
 }
 
+// filterCLIUpgrades filters out CLI versions that are not valid upgrades.
+func filterCLIUpgrades(currentVersion string, newVersions []string) []string {
+	result := []string{}
+	for i := range newVersions {
+		if err := compatibility.IsValidUpgrade(currentVersion, newVersions[i]); err != nil {
+			fmt.Println(err)
```

Comment:
User should not see all those incompatible upgrades.
Maybe log them to debug output.